warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableExpansionVisitor.java

```diff
@@ -0,0 +1,437 @@
+    /**
+     * Distribute a parent andNode into the orNode, using the andNodeChild to determine which children of the andNode should be distributed and which contain
+     * the orNode. orNode should have been passed to canExpand() and returned true prior to calling.
+     *
+     * @param andNode
+     *            the first andNode parent from the orNode
+     * @param andNodeChild
+     *            the direct child of the andNode that is also a parent of orNode
+     * @param orNode
+     *            the orNode to distribute the andNode into
+     * @return a new orNode to be substituted for the andNode in the original query tree with a logically equivalent (and executable) state
+     * @throws IllegalStateException
+     *             if a single node path can't be traced between the andNode and orNode
+     */
+    private ASTOrNode distribute(ASTAndNode andNode, JexlNode andNodeChild, ASTOrNode orNode) {
+        // hold the list of nodes that are to be distributed into the orNode grab everything in the and that isn't the branch to be expanded into
+        List<JexlNode> otherNodes = new ArrayList<>(andNode.jjtGetNumChildren() - 1);
+        for (int i = 0; i < andNode.jjtGetNumChildren(); i++) {
+            JexlNode candidate = andNode.jjtGetChild(i);
+            if (candidate != andNodeChild) {
+                otherNodes.add(candidate);
+            }
+        }
+        
+        // create a top level orNode to join together the distributed elements
+        ASTOrNode newOr = new ASTOrNode(ParserTreeConstants.JJTORNODE);
+        
+        // since only ASTReference and ASTReferenceExpression nodes are allowed between the andNode and orNode all the children which should always have a
+        // single child check if there is a set of nodes between the child and orNode that need to be passed along
+        if (andNodeChild == orNode) {
+            // clear the child, its the same as the orNode and there is nothing to bridge
+            andNodeChild = null;
+        } else if (orNode.jjtGetParent().jjtGetNumChildren() == 1) {
+            // replace the orNode with a null so it won't be copied below
+            JexlNodes.children(orNode.jjtGetParent(), new JexlNode[0]);
+        } else {
+            // log an error and abort the distribute, this should never happen
+            log.warn("Unexpected number of children on the orNode parent, expected 1 got " + orNode.jjtGetParent().jjtGetNumChildren() + " aborting distribute");
+            throw new IllegalStateException("Unexpected number of children on the orNode parent, expected 1 got " + orNode.jjtGetParent().jjtGetNumChildren()
+                            + " aborting distribute");
+        }
+        
+        // fetch the minimal expansion terms instead of expanding everything
+        List<JexlNode> orTerms = getOrTerms(orNode, andNodeChild);
+        
+        // everything goes together, abort
+        if (orTerms.size() == 1) {
+            return null;
+        }
```

Comment:
This is the only change besides unit tests

Reply:
Okey dokey.  Thanks for pointing that out.